tests/tests/wgpu-gpu/samplers.rs

```diff
@@ -59,8 +62,15 @@ fn sampler_deduplication(ctx: TestingContext) {
 }
 
 #[gpu_test]
-static SAMPLER_CREATION_FAILURE: GpuTestConfiguration =
-    GpuTestConfiguration::new().run_sync(sampler_creation_failure);
+static SAMPLER_CREATION_FAILURE: GpuTestConfiguration = GpuTestConfiguration::new()
+    .parameters(
+        // Removal of this exception is tracked at <https://github.com/gfx-rs/wgpu/issues/7805>.
+        TestParameters::default().expect_fail(FailureCase::backend_adapter(
+            Backends::DX12,
+            "Microsoft Basic Render Driver",
+        )),
```

Comment:
This will start failing with a device lost error type on the D3D12 backend in general, not just WARP.

Can we keep the existing behavior where this returns an OOM error type instead?
At least until we have a resolution on https://github.com/gpuweb/gpuweb/issues/5142.

Reply:
Actually, looking at the changes in this PR it's not clear to me why the test is now failing since we are still mapping the HAL OOM error to an OOM error type.

Reply:
But if that's the case, why is this test succeeding on macOS and Linux? Shouldn't we be expecting the same type of error across all platforms?

Reply:
It works! Filed 8e0d4dff and a refactor to support it as a parent commit. WDYT, @teoxoy?

…what a dumb test, to only fail on one platform. 😭